src/main/java/net/masterthought/cucumber/generators/FailuresOverviewPage.java

```diff
@@ -16,6 +21,19 @@ public String getWebPage() {
 
     @Override
     public void prepareReport() {
-        context.put("all_features", report.getAllFeatures());
+        List<Element> failures = new ArrayList<>();
+        for (Feature feature : report.getAllFeatures()) {
+            if (feature.getStatus().isPassed()) continue;
+
+            for (Element element : feature.getElements()) {
+                if (!element.getBeforeStatus().isPassed()
+                    || !element.getStepsStatus().isPassed()
+                    || !element.getAfterStatus().isPassed()) {
```

Comment:
In that case if the Before or After is failed but the scenario passes you will print the element even it is passed.

Is that what you expected?